explore/index.html

```diff
@@ -26,7 +26,7 @@ <h2 class="header-red">OME for Scientists</h2>
                 <p>Sign up for our <a href="http://help.openmicroscopy.org/demo-server.html">demo server</a> to test out the functionality with your own data.</p>
```

Comment:
Left this link pointing at the Help page because it's more for info rather than specifically saying 'Sign up here'